### PR TITLE
Reads from the CT Server (if set)

### PR DESCRIPTION
**Pull request by AndrewGorton**
Uses _&lt;register&gt;_.source to dictate if it uses PostGres or CTServer.

### CODE REVIEW

File: src/main/java/uk/gov/indexer/ctserver/Audit.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package uk.gov.indexer.ctserver;
+
+import java.util.List;
+
+public class Audit {
```
Comment (philandstuff) [inline src/main/java/uk/gov/indexer/ctserver/Audit.java]: I'm not sure Audit is the right name for this -- based on the [endpoint name & docs](http://tools.ietf.org/html/rfc6962#section-4.5), maybe "Proof" or "AuditProof"?

File: src/main/java/uk/gov/indexer/ctserver/CTServer.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+import java.io.ByteArrayOutputStream;
+import java.security.MessageDigest;
+import java.util.Base64;
+
+public class CTServer {
```
Comment (philandstuff) [inline src/main/java/uk/gov/indexer/ctserver/CTServer.java:11]: I wonder if, longer term, we'll want to extract this into a client library? Not something for this PR anyway.

File: src/main/java/uk/gov/indexer/Indexer.java [hunk 3]
```diff
@@ -21,21 +28,29 @@ public Indexer(Configuration configuration, String register) {
...
             destinationDBUpdateDAO = destDbi.open().attach(DestinationDBUpdateDAO.class);
 
-            sourceDBQueryDAO = sourceDbi.onDemand(SourceDBQueryDAO.class);
-            executorService.scheduleAtFixedRate(
-                    new IndexerTask(register, sourceDBQueryDAO, destinationDBUpdateDAO),
-                    0,
-                    10,
-                    TimeUnit.SECONDS
-            );
+            if (configuration.getCTServerEndpointForRegister(register).isPresent()) {
```
Comment (om-sharma) [inline src/main/java/uk/gov/indexer/Indexer.java]: can be simplified as below: 

           Fetcher fetcher = configuration.getCTServerEndpointForRegister(register)
                    .map(endPoint ->
                                    (Fetcher) new CTFetcher(new CTServer(endPoint))
                    ).orElse(
                            new PGFetcher(sourceDBQueryDAO)
                    );

            executorService.scheduleAtFixedRate(
                    new IndexerTask(register, fetcher, destinationDBUpdateDAO),
                    0,
                    10,
                    TimeUnit.SECONDS);

File: src/main/java/uk/gov/indexer/ctserver/CTServer.java (hunk 2 above)
Comment (philandstuff) [inline src/main/java/uk/gov/indexer/ctserver/CTServer.java:11]: Mentioned in slack but recording here for posterity: there already is a [java ct client](https://github.com/google/certificate-transparency/blob/master/java/src/org/certificatetransparency/ctlog/comm/HttpLogClient.java) but it's not clear how to build it with the gclient workflow and I don't think any jars are published for it.

EDIT: this is just for information, not something blocking the PR or anything.

File: src/main/java/uk/gov/indexer/ctserver/AuditProofs.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package uk.gov.indexer.ctserver;
+
+import java.util.List;
+
+public class AuditProofs {
```
Comment (philandstuff) [inline src/main/java/uk/gov/indexer/ctserver/AuditProofs.java]: nitpick: I think this is a single AuditProof, not a collection of many AuditProofs

**Comment by philandstuff** [comment]
@om-sharma is there anything holding this back?

**Pull request merged**